server/sonar-server/src/main/java/org/sonar/server/component/es/ProjectMeasuresResultSetIterator.java

```diff
@@ -32,32 +34,54 @@
 import org.sonar.db.DbSession;
 import org.sonar.db.ResultSetIterator;
 
-public class ProjectMeasuresResultSetIterator extends ResultSetIterator<ProjectMeasuresDoc> {
+import static org.sonar.api.measures.Metric.ValueType.DATA;
+import static org.sonar.api.measures.Metric.ValueType.DISTRIB;
+import static org.sonar.db.DatabaseUtils.repeatCondition;
 
-  private static final String[] FIELDS = {
-    "p.uuid",
-    "p.kee",
-    "p.name",
-    "s.created_at"
-  };
+public class ProjectMeasuresResultSetIterator extends ResultSetIterator<ProjectMeasuresDoc> {
 
-  private static final String SQL_ALL = "SELECT " + StringUtils.join(FIELDS, ",") + " FROM projects p " +
+  private static final String SQL_PROJECTS = "SELECT p.uuid, p.kee, p.name, s.uuid, s.created_at FROM projects p " +
     "LEFT OUTER JOIN snapshots s ON s.component_uuid=p.uuid AND s.islast=? " +
     "WHERE p.enabled=? AND p.scope=? AND p.qualifier=?";
 
   private static final String DATE_FILTER = " AND s.created_at>?";
 
-  private static final String PROJECT_FILTER = " AND p.uuid=?";
+  private static final String FILTER_PROJECT = " AND p.uuid=?";
+
+  private static final String SQL_METRICS = "SELECT m.id, m.name FROM metrics m " +
+    "WHERE m.val_type NOT IN ('" + StringUtils.join(new String[] {DATA.name(), DISTRIB.name()}, "','") + "') " +
```

Comment:
I think I'd prefer a whitelist than a black one. If we add a new type that is a data type, it will fail